posthog/hogql/printer.py

```diff
@@ -1015,6 +1015,32 @@ def visit_call(self, node: ast.Call):
         if optimized_property_group_call := self.__get_optimized_property_group_call(node):
             return optimized_property_group_call
 
+        # Handle format strings in function names before checking function type
+        if func_meta := (
+            find_hogql_aggregation(node.name)
+            or find_hogql_function(node.name)
+            or find_hogql_posthog_function(node.name)
+        ):
+            if "{" in func_meta.clickhouse_name:
+                # Check if using positional arguments (e.g. {0}, {1})
+                if any(f"{{{i}}}" in func_meta.clickhouse_name for i in range(10)):  # Check common positions 0-9
```

Comment:
I'd much rather we had a setting on `HogQLFunctionMeta` to indicate whether we're using positional args than trying to string match here. Likewise a property for whether formatting is even required